project/FormatStagedScalaFilesPlugin.scala

```diff
@@ -0,0 +1,76 @@
+import better.files.File
+import org.scalafmt.sbt.ScalafmtPlugin
+import sbt.Keys.streams
+import sbt.nio.Keys.{ReloadOnSourceChanges, onChangedBuildSource}
+import sbt.{Compile, Global, Setting, taskKey}
+import utils.Step
+
+object FormatStagedScalaFilesPlugin extends sbt.AutoPlugin {
+  override def trigger = allRequirements
+
+  object autoImport {
+    val formatStagedScalaFiles = taskKey[Unit]("Format staged Scala files")
+  }
+
+  import autoImport._
+
+  override def projectSettings = Seq(
+    formatStagedScalaFiles in Global := {
+      formatStagedScalaFilesOnly().value
+    }
+  )
+
+  override def globalSettings: Seq[Setting[_]] = Seq(
+    Global / onChangedBuildSource := ReloadOnSourceChanges,
+  )
+
+  private def formatStagedScalaFilesOnly() = {
+    val result = for {
+      stagedFiles <- getStagedScalaFiles()
+      _ <-
+        if (stagedFiles.nonEmpty) {
+          for {
+            _ <- callFormatFiles(stagedFiles)
+            _ <- addToGitAllStagedFilesOnceAgain(stagedFiles)
+          } yield ()
+        } else {
+          Step.taskUnit
+        }
+    } yield ()
+    result.runThrowing
+  }
+
+  private def getStagedScalaFiles() = Step.deferredTask {
+    os
+      .proc("git", "diff", "--cached", "--name-only", "--diff-filter=ACM")
+      .call()
+      .out
+      .lines()
+      .filter(f => f.endsWith(".scala") || f.endsWith(".sbt"))
+      .toList
+  }
+
+  private def callFormatFiles(files: List[String]) = {
+    val userDir       = getUserDir()
+    val fullPathFiles = files.map(f => s"${(userDir / f).path}")
```

Comment:
Do we need absolute paths here? If so, can we get root path from `baseDirectory.value`, or we need to use Java's system property?

Reply:
I've added the plugin explicitly to the root module of the project and now `baseDirectory` works as expected

Reply:
Then shouldn't you also remove absolute path resolution? `scalafmtOnly` does it for you

Reply:
yes, yes. The review fixes were not ready yet. They are now.